app/utils/markdown/index.ts

```diff
@@ -119,71 +127,97 @@ export const getMarkdownBlockStyles = makeStyleSheetFromTheme((theme: Theme) =>
+    lua: {name: 'Lua', extensions: ['lua']},
+    makefile: {name: 'Makefile', extensions: ['mk', 'mak'], aliases: ['make', 'mf', 'gnumake', 'bsdmake']},
+    markdown: {name: 'Markdown', extensions: ['md', 'mkdown', 'mkd'], aliases: ['md', 'mkd']},
+    matlab: {name: 'Matlab', extensions: ['matlab', 'm'], aliases: ['m']},
+    objectivec: {name: 'Objective C', extensions: ['mm', 'objc', 'obj-c'], aliases: ['objective_c', 'objc']},
+    ocaml: {name: 'OCaml', extensions: ['ml']},
+    perl: {name: 'Perl', extensions: ['perl', 'pl'], aliases: ['pl']},
+    pgsql: {name: 'PostgreSQL', extensions: ['pgsql', 'postgres', 'postgresql'], aliases: ['postgres', 'postgresql']},
+    php: {name: 'PHP', extensions: ['php', 'php3', 'php4', 'php5', 'php6'], aliases: ['php3', 'php4', 'php5']},
+    powershell: {name: 'PowerShell', extensions: ['ps', 'ps1'], aliases: ['posh']},
+    puppet: {name: 'Puppet', extensions: ['pp'], aliases: ['pp']},
+    python: {name: 'Python', extensions: ['py', 'gyp'], aliases: ['py']},
+    r: {name: 'R', extensions: ['r'], aliases: ['r', 's']},
+    ruby: {name: 'Ruby', extensions: ['ruby', 'rb', 'gemspec', 'podspec', 'thor', 'irb'], aliases: ['rb']},
+    rust: {name: 'Rust', extensions: ['rs'], aliases: ['rs']},
+    scala: {name: 'Scala', extensions: ['scala']},
+    scheme: {name: 'Scheme', extensions: ['scm', 'sld']},
+    scss: {name: 'SCSS', extensions: ['scss']},
+    smalltalk: {name: 'Smalltalk', extensions: ['st'], aliases: ['st', 'squeak']},
+    sql: {name: 'SQL', extensions: ['sql']},
+    stylus: {name: 'Stylus', extensions: ['styl'], aliases: ['styl']},
+    swift: {name: 'Swift', extensions: ['swift']},
+    text: {name: 'Text', extensions: ['txt', 'log']},
+    typescript: {name: 'TypeScript', extensions: ['ts', 'tsx'], aliases: ['ts', 'tsx']},
+    vbnet: {name: 'VB.Net', extensions: ['vbnet', 'vb', 'bas'], aliases: ['vb', 'visualbasic']},
+    vbscript: {name: 'VBScript', extensions: ['vbs']},
+    verilog: {name: 'Verilog', extensions: ['v', 'veo', 'sv', 'svh']},
+    vhdl: {name: 'VHDL', extensions: ['vhd', 'vhdl']},
+    xml: {name: 'HTML, XML', extensions: ['xml', 'html', 'xhtml', 'rss', 'atom', 'xsl', 'plist']},
+    yaml: {name: 'YAML', extensions: ['yaml'], aliases: ['yml']},
 };
 
-export function getDisplayNameForLanguage(language: string) {
-    return languages[language.toLowerCase()] || '';
+function canHighlight(language: string): boolean {
+    return Boolean(getHighlightLanguageFromNameOrAlias(language));
+}
+
+export function getHighlightLanguageFromNameOrAlias(name: string) {
+    const langName: string = name.toLowerCase();
+    if (highlightedLanguages[langName]) {
+        return langName;
+    }
+
+    return Object.keys(highlightedLanguages).find((key) => {
+        const aliases = new Set(highlightedLanguages[key].aliases);
+        return aliases.has(langName);
+    }) || '';
+}
+
+export function getHighlightLanguageName(language: string): string {
+    if (canHighlight(language)) {
```

Comment:
This function calls `getHighlightLanguageFromNameOrAlias` so it can do it again in the next line. And you are already doing the check on whether you have or not the name.

Reply:
again copied from webapp